src/frontend/web_application/server/middleware/ssr.js

```diff
@@ -31,10 +31,16 @@ function getMarkup(reactElement, store, assets) {
   ].reduce((str, current) => str.replace(current.key, current.value), tpl);
 }
 
+function applyUserLocaleToGlobal(req) {
+  global.USER_LOCALE = req.cookies.locale || req.locale;
```

Comment:
no it can't be in `initialState` since react-translate does not officially load a locale and a translation after the app boostrap.

req.locale exists, it is a middleware defined in main express file. I don't knw how to be more explicit right now